### ISSUE TITLE
pod update/install did not fetch framework as in podfile

### ISSUE THREAD

**Issue by littlepod** [CocoaPods/CocoaPods#6402]
# Report

## What did you do?
After lots of reading and trial I am posting this for help.

Just like "google-cast-sdk" (https://github.com/CocoaPods/Specs/blob/master/Specs/8/1/2/google-cast-sdk/3.3.0/google-cast-sdk.podspec.json), I am trying to include framework as pod to my project. All is to explore feasibility, if I can create pod out of my frameworks and include to my project directly by a simple "pod install".

My s.source file is a ZIP file at http location as like "google-cast-sdk". I am using "s.vendored_frameworks" to have framework.

Below is my **podspec** file snippet (only main stuff listed)

```ruby
s.source       = {:http => "http://sdf.samsungcloudcdn.com/Public/UwBWAEMAMAAwADAAMAAyAA==/MwA3ADgANQBvAGEAYQA1ADkANgBlADcAZwA=/SmartViewSDK/SmartViewSDK-iOS-XCode8.1-2.3.8.zip" }
s.source_files  = "SmartViewSDK-iOS", "SmartViewSDK-iOS/iphoneos+iphonesimulator/SmartView.framework/Headers/*.h"
s.preserve_paths    = "SmartViewSDK-iOS/iphoneos+iphonesimulator/SmartView.framework"
s.vendored_frameworks   = "SmartViewSDK-iOS/iphoneos+iphonesimulator/SmartView.framework"
```

My podspec file is good as "pod spec lint" is successful without any problem.
Even I checked my cache for this pod and I see framework there (./Library/Caches/CocoaPods/Pods/External/SmartViewSDK-iOS/3031ec9b173d86f80c7871a46ab3193b/...)

Below is my **podfile** file for a sample project taken from net

```ruby
inhibit_all_warnings!
use_frameworks!
target 'Phonercise' do
    pod 'SmartViewSDK-iOS', :git =>'https://github.com/littlepod/DemoSmartPod.git'
end
```

now, when I do `pod install`

I do not get "SmartViewSDK.framework" file in "../Phonercise/Pods/SmartViewSDK-iOS" as like I do get for google-cast-sdk. Rather I get two general README and LICENSE files that is there on my git repository where my pod spec lies.

## What did you expect to happen?

I was expecting to see framework in my project path.
 

## What happened instead?

wo general README and LICENSE files that is there on my git repository where my pod spec lies.

## CocoaPods Environment

ocoaPods : 1.1.1
        Ruby : ruby 2.2.5p319 (2016-04-26 revision 54774) [x86_64-darwin15]
    RubyGems : 2.4.8
        Host : Mac OS X 10.11.5 (15F34)
       Xcode : 8.0 (8A218a)
         Git : git version 2.8.4 (Apple Git-73)
Ruby lib dir : /Users/ankita/.rvm/rubies/ruby-2.2.5/lib
Repositories : littlepod - https://github.com/littlepod/DemoSmartPod.git @ 8a43480a88d69ff8f3bf918a470d99aea9c126fb
               master - https://github.com/CocoaPods/Specs.git @ 1a0138a3e5c385faa9d7f429dce4ab82888c32b2
```

### Installation Source

```
Executable Path: /Users/ankita/.rvm/rubies/ruby-2.2.5/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.1.2
cocoapods-try         : 1.1.0

## Project that demonstrates the issue
https://github.com/littlepod/DemoSmartPod

Please help to resolve this.

Thanks,
Om

**Comment by Noobish1** [comment]
I've seen this happen as well. When trying to add Swiftlint to our pod using:
```
pod 'SwiftLint', :git => 'https://github.com/realm/SwiftLint.git'
```

it ends up just downloading the repo as a zip rather than actually downloading from the provided link, I've tried placing completely different URLs as the source and it'll always ignore it.

**Comment by benasher44** [comment]
Hi there! Thanks for the report! I've tried to repri this, but I'm getting an error. It's looking for SmartViewSDK-iOS.podspec in your sample project, but it's named something else. Can you update the sample project? I'd be happy to give it another go once that's done! Thanks!

**Comment by littlepod** [comment]
I was able to resolve this issue. Mistake was at my end.

Since this was private pod, I needed to add and link my private pod with git url with the steps given in url
[](https://guides.cocoapods.org/making/private-cocoapods.html)

$ pod repo add REPO_NAME SOURCE_URL

PS: Don't try to use my previous pod reference and path. Its obsolete now as that was a trial one.

**Request you to close this query or remove it.**

Thanks
